csi.proto

```diff
@@ -264,6 +269,21 @@ message VolumeMetadata {
   // each Plugin keeps this information as small as possible.
   map<string, string> values = 1;
 }
+
+// A standard way to encode credential data. The total bytes of the values in
+// the Data field must be less than 1 Mebibyte.
+message Credentials {
+  // Data contains the credential data, for example username and password.
+  // Each key must consist of alphanumeric characters, '-', '_' or '.'.
+  // Each value MUST contain a valid string. An SP MAY choose to accept binary
+  // (non-string) data by using a binary-to-text encoding scheme, like base64.
+  // An SP SHALL advertise the requirements for credentials in documentation.
+  // COs SHALL permit users to pass through the required credentials.
+  // This information is sensitive and MUST be treated as such (not logged,
+  // etc.) by the CO.
+  // This field is REQUIRED.
+  map<string, string> Data = 1;
```

Comment:
@saad-ali that all sounds good. I do have one question: if a secret store only stores binary data (because it does not care about the type of the secret it stores), when the CO retrieves the secret from the secret store, it needs to convert the binary secret to string format. What the CO needs to do here?

Or we dictate that the secret store the CO uses must support string format so that CO can just pass those secrets to the plugin?